packages/medusa/src/api/routes/store/collections/list-collections.ts

```diff
@@ -0,0 +1,57 @@
+import { Type } from "class-transformer"
+import { IsOptional } from "class-validator"
+import ProductCollectionService from "../../../../services/product-collection"
+import { validator } from "../../../../utils/validator"
+/**
+ * @oas [get] /collections
+ * operationId: "GetCollections"
+ * summary: "List Product Collections"
+ * description: "Retrieve a list of Product Collection."
+ * parameters:
+ *   - (query) offset=0 {integer} The number of collections to skip before starting to collect the collections set
+ *   - (query) limit=10 {integer} The number of collections to return
+ * tags:
+ *   - Collection
+ * responses:
+ *  "200":
+ *    description: OK
+ *    content:
+ *      application/json:
+ *        schema:
+ *          properties:
+ *            collection:
+ *              $ref: "#/components/schemas/product_collection"
+ */
+export default async (req, res) => {
+  const { limit, offset } = await validator(
+    StoreGetCollectionsParams,
+    req.query
+  )
+  const selector = {}
+
+  const productCollectionService: ProductCollectionService = req.scope.resolve(
+    "productCollectionService"
+  )
+
+  const listConfig = {
+    skip: offset,
+    take: limit,
+  }
+
+  const [collections, count] = await productCollectionService.listAndCount(
+    selector,
+    listConfig
+  )
+
+  res.status(200).json({ collections, count, limit, offset })
+}
+
+export class StoreGetCollectionsParams {
+  @IsOptional()
+  @Type(() => Number)
+  limit?: number = 10
```

Comment:
Question; I do not know if necessary (but we do it other places): add `@IsInt()` to both of these?

Reply:
Thanks for pointing it out! Adding `@IsInt()` is pretty important as it helps prevent query params such as `?offset=bla&limit=blo` from going down the service layer :+1: